src/components/ContentPreview/ContentPreview.js

```diff
@@ -612,7 +623,27 @@ class ContentPreview extends PureComponent<Props, State> {
     fetchFileSuccessCallback = (file: BoxItem): void => {
         this.fetchFileEndTime = performance.now();
         this.retryCount = 0;
-        this.setState({ file, isFileError: false });
+
+        const { file: currentFile }: State = this.state;
+        const isExistingFile = currentFile ? currentFile.id === file.id : false;
+        const isWatermarked = getProp(file, 'watermark_info.is_watermarked', false);
+
+        // If the file is watermarked or if its a new file, then update the state
+        // In this case preview should reload without prompting the user
+        if (isWatermarked || !isExistingFile) {
+            this.setState({ file, isFileError: false, isReloadNotificationVisible: false });
+            return;
+        }
+
+        // If we are already prevewing the file that got updated then show the
+        // user a notification to reload the file only if its sha1 changed
+        // $FlowFixMe file version and sha1 should exist at this point
+        if (currentFile.file_version.sha1 !== file.file_version.sha1) {
```

Comment:
Save this boolean in a variable and then set the state with it's result? Can get rid of the else case.